src/mds/MDSRank.cc

```diff
@@ -2497,6 +2499,42 @@ bool MDSRank::command_dirfrag_ls(
   return true;
 }
 
+void MDSRank::command_openfiles_ls(Formatter *f) 
+{
+  Mutex::Locker l(mds_lock);
+  
+  for (auto p : mdlog->get_segments()) {
+    LogSegment *ls = p.second;
+    auto q = ls->open_files.begin(member_offset(CInode, item_open_file));
+    while (!q.end()) {
+      CInode *in = *q;
+      ++q;
+      if (in->last == CEPH_NOSNAP) {
+        if (in->is_any_caps_wanted()) {
+          std::string path;
+          in->make_path_string(path, true);
+          if (path.empty()) {  // empty path means it's root, use "/" for readability 
+            path = "/";
+          }
```

Comment:
Here, I would suggest dumping every inode as:

````
{
  "path": "<path>",
  "inode": CInode::dump()
}
````

Then you don't need to dump the caps below:

Reply:
agree, that's more better

Reply:
@batrick  my intention is just focus on how clients are trying to access the opened files,so i only dumped the caps here since CInode::dump() is just giving too much info may full over the screen

Reply:
That's reasonable, perhaps a flag argument to CInode::dump() to only show certain information then. (e.g. flag for the inode number, flag for caps, flag for full information). You may also want to expose the flags to the admin socket commands or just use it internally for "openfiles ls". Whatever you feel like doing.

Reply:
compact inode info make sense for me

Reply:
thank you, that's more clean, make sense for me too